internal/characters/yaoyao/yuegui.go

```diff
@@ -0,0 +1,179 @@
+	// yg.Reactable = &reactable.Reactable{}
+	// yg.Reactable.Init(yg, c.Core)
+	yg.aoe = combat.NewCircleHitOnTarget(pos, nil, skillTargetingRad)
+
+	return yg
+}
+
+func (c *char) newYueguiJump() {
+	if !c.StatusIsActive(burstKey) || c.numYueguiJumping >= 3 {
+		return
+	}
+	yg := &yuegui{
+		ai:   c.burstRadishAI,
+		snap: c.Snapshot(&c.burstRadishAI),
+		c:    c,
+	}
+	player := c.Core.Combat.Player()
+	pos := geometry.CalcOffsetPoint(player.Pos(), geometry.Point{Y: -2}, player.Direction())
+	yg.Gadget = gadget.New(c.Core, pos, 0.5, combat.GadgetTypYueguiJumping)
+	yg.Gadget.Duration = -1 // They last until they get deleted by the burst
+	yg.Gadget.OnThinkInterval = yg.throw
+
+	// they start throwing 29f after being spawned
+	yg.Gadget.ThinkInterval = 29
+
+	yg.Gadget.OnKill = func() {
+		yg.Core.Log.NewEvent("Yuegui (Jumping) removed", glog.LogCharacterEvent, yg.c.Index)
+	}
+	yg.Core.Log.NewEvent("Yuegui (Jumping) summoned", glog.LogCharacterEvent, yg.c.Index)
+	// yg.Reactable = &reactable.Reactable{}
+	// yg.Reactable.Init(yg, c.Core)
+	yg.aoe = combat.NewCircleHitOnTarget(pos, nil, skillTargetingRad)
+
+	c.Core.Combat.AddGadget(yg)
+	c.yueguiJumping[c.numYueguiJumping] = yg
+	c.numYueguiJumping += 1
+}
+
+func (c *char) makeHealCB(area combat.AttackPattern, hi player.HealInfo) func(combat.AttackCB) {
+	done := false
+	return func(a combat.AttackCB) {
+		if a.Target.Type() != targets.TargettableEnemy {
+			return
+		}
+
+		if done {
+			return
+		}
+		if c.Core.Combat.Player().IsWithinArea(area) {
+			c.radishHeal(hi)
+			done = true
+		}
+	}
+}
+
+func (yg *yuegui) Tick() {
+	//this is needed since both reactable and gadget tick
+	// yg.Reactable.Tick()
+	yg.Gadget.Tick()
+}
+
+func (yg *yuegui) makeParticleCB() combat.AttackCBFunc {
+	if yg.GadgetTyp() != combat.GadgetTypYueguiThrowing {
+		return nil
+	}
+	return func(a combat.AttackCB) {
+		if a.Target.Type() != targets.TargettableEnemy {
+			return
+		}
+
+		if yg.c.StatusIsActive(skillParticleICD) {
+			return
+		}
+		yg.c.AddStatus(skillParticleICD, 1.5*60, true)
+		yg.Core.QueueParticle(yg.c.Base.Key.String(), 1, attributes.Dendro, yg.c.ParticleDelay)
+	}
+}
+
+func (yg *yuegui) throw() {
+	yg.Gadget.ThinkInterval = 60
+	currHPPerc := yg.Core.Player.ActiveChar().HPCurrent / yg.Core.Player.ActiveChar().MaxHP()
+	enemy := yg.Core.Combat.RandomEnemyWithinArea(yg.aoe, nil)
+
+	var target geometry.Point
+	if currHPPerc > 0.7 && enemy != nil {
+		target = enemy.Pos()
+	} else {
+		// really it should be random if no targets are in range and the character's HP is full but we aren't really simming that
+		target = yg.Core.Combat.Player().Pos()
+	}
+	hi := yg.getHeal()
+	radishExplodeAoE := combat.NewCircleHitOnTarget(target, nil, radishRad)
+
+	yg.Core.QueueAttackWithSnap(
+		yg.ai,
+		yg.snap,
+		radishExplodeAoE,
+		travelDelay,
+		yg.c.makeHealCB(radishExplodeAoE, hi),
+		yg.makeParticleCB(),
+		yg.c.makeC2CB(),
+	)
```

Comment:
You're missing the mechanic that the ai and hi is decided on explosion, so if the Yuegui (Throwing) radish hits an enemy when the burst is up, then the radish is burst dmg, has the burst icd and burst healing scaling. Snapshot doesn't change here.

Reply:
The same mechanic should apply to the A1 radish.

Reply:
???